src/figuresClasses.ts

```diff
@@ -1,11 +1,69 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid triangle sides');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    return +Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c)).toFixed(
+      2,
+    );
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  public shape: 'circle' = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Invalid circle radius');
```

Comment:
The error message for the `Circle` constructor should be more specific. Consider changing it to something like 'Radius must be greater than zero.' to provide a clearer description of the issue.